ext/src/helpers/output_limited_buffer.rs

```diff
@@ -21,8 +75,12 @@ impl OutputLimitedBuffer {
     }
 }
 
-impl io::Write for OutputLimitedBuffer {
-    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+impl OutputLimitedBuffer {
```

Comment:
We still have to use a custom type since the default in-memory implementation will trap when its capacity is exceeded and it looks like we don't want to trap when the capacity for our implementation is exceeded